src/ripple/app/main/Main.cpp

```diff
@@ -37,30 +37,39 @@
 #include <ripple/beast/clock/basic_seconds_clock.h>
 #include <ripple/beast/core/CurrentThreadName.h>
 
-#include <beast/unit_test/dstream.hpp>
-#include <beast/unit_test/global_suites.hpp>
 #include <beast/unit_test/match.hpp>
-#include <beast/unit_test/reporter.hpp>
 #include <test/unit_test/multi_runner.h>
 
 #include <google/protobuf/stubs/common.h>
 
 #include <boost/filesystem.hpp>
 #include <boost/process.hpp>
 #include <boost/program_options.hpp>
-#include <boost/system/error_code.hpp>
-#include <boost/version.hpp>
+#include <boost/predef.h>
 
 #include <cstdlib>
 #include <iostream>
 #include <utility>
 #include <stdexcept>
 
-#ifdef _MSC_VER
+#if BOOST_OS_WINDOWS
 #include <sys/types.h>
 #include <sys/timeb.h>
 #endif
 
+// Do we know the plaform we're compiling on? If you're adding new platforms
+// modify this check accordingly.
+#if !BOOST_OS_LINUX && !BOOST_OS_WINDOWS && !BOOST_OS_MACOS
+#error Supported platforms are: Linux, Windows and MacOS
+#endif
+
+// Ensure that precisely one platform is detected.
+#if (BOOST_OS_LINUX   && (BOOST_OS_WINDOWS || BOOST_OS_MACOS)) || \
+    (BOOST_OS_MACOS   && (BOOST_OS_WINDOWS || BOOST_OS_LINUX)) || \
+    (BOOST_OS_WINDOWS && (BOOST_OS_LINUX   || BOOST_OS_MACOS))
+#error Multiple supported platforms appear active at once
+#endif
```

Comment:
At the risk of bikeshedding, what do you think about
```
constexpr int boost_os_check = []() {
    size_t check{0};
#if BOOST_OS_LINUX
    ++check;
#endif
#if BOOST_OS_WINDOWS
    ++check;
#endif
#if BOOST_OS_MACOS
    ++check;
#endif
    return check;
}();
static_assert(
    boost_os_check > 0,
    "Supported platforms are: Linux, Windows and MacOS");
static_assert(
    boost_os_check < 2,
    "Multiple supported platforms appear active at once");
```
It would make adding future platforms a little bit easier.

Reply:
Looks reasonable, but if it's all the same to you, I think I'll punt on this for now.

Reply:
Not a problem.